src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -0,0 +1,257 @@
+        assert "Accepted password for foo4" in secure_log_str, "Accepted password not found in /var/log/secure"
+
+        assert f"Access denied for user [foo3@{ds_instance_name}]" in sssd_log_str, \
+               "Access denied not found in /var/log/sssd/sssd_example1.log"
+        assert f"Access allowed for user [foo4@{ds_instance_name}]" in sssd_log_str, \
+               "Access allowed not found in /var/log/sssd/sssd_example1.log"
+
+    @staticmethod
+    def test_0002_k5login_user3(multihost):
+        """
+        :title: .k5login file has user principal for foo3
+        :id: decede10-c500-4b56-8037-7f2e325009b6
+        :setup:
+          1. Clear the secure log, clear cache and restart sssd.
+          2. Add the foo3 user principal in the k5login file
+        :steps:
+          1. Authenticate the user foo3 from the client
+          2. Check the secure log for expected messages.
+          3. Check the sssd domain log for expected messages.
+        :expectedresults:
+          1. User foo3 should be able to successfully login
+          2. Secure Log contains the expected lines:
+             pam_sss(sshd:auth): authentication success for foo3
+             Accepted password for foo3
+          3. SSSD Domain Log contains the expected lines:
+             Access allowed for user [foo3
+        :teardown:
+          1. Clear the .k5login file.
+        """
+        multihost.client[0].run_command("truncate -s 0 /var/log/secure", raiseonerr=False)
+        client = sssdTools(multihost.client[0])
+        client.clear_sssd_cache()
+
+        cmds = ['echo "foo3@EXAMPLE.TEST" > /home/foo3/.k5login',
+                "restorecon -v /home/foo3/.k5login"]
+        for command in cmds:
+            cmd = multihost.client[0].run_command(command, raiseonerr=False)
+
+        user = 'foo3'
+        ssh = client.auth_from_client(user, 'Secret123')
+
+        secure_log_str = multihost.client[0].get_file_contents("/var/log/secure").decode('utf-8')
+
+        file = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+        sssd_log_str = multihost.client[0].get_file_contents(file).decode('utf-8')
+
+        multihost.client[0].run_command("truncate -s 0 /home/foo3/.k5login", raiseonerr=False)
+        assert cmd.returncode == 0, f"{command} did not execute successfully"
+        assert ssh == 3, "foo3 failed to log in"
+        assert "pam_sss(sshd:auth): authentication success" in secure_log_str, \
+               "authentication success not found in /var/log/secure"
+        assert "Accepted password for foo3" in secure_log_str, \
+               "Accepted password not found in /var/log/secure"
+        assert f"Access allowed for user [foo3@{ds_instance_name}]" in sssd_log_str, \
+               "Access allowed not found in /var/log/sssd/sssd_example1.log"
+
+    @staticmethod
+    def test_0003_k5login_user3_with_user4(multihost):
+        """
+        :title: .k5login file has user principals for user foo3 and foo4.
```

Comment:
IDM-SSSD-TC: krb_provider: krb_access_provider: k5login has testuser3 and testuser4